src/app/dts/template-editor/template-editor.component.html

```diff
@@ -32,23 +32,71 @@
         {{ statusMessage }}
       </span>
     </div>
-    <button id="saveButton" type="submit" [disabled]="!canSave()" class="push" mat-button>
+    <button id="saveButton" type="submit" [disabled]="!canSave()" mat-button>
       Save
     </button>
+    <button type="button" (click)="toggleDataFields()" class="push" mat-button>
+      Fields
+    </button>
   </div>
+  <div class="flex-row">
+    <div style="flex: 1">
+      <!-- CKEditor control -->
+      <div id="editor1" name="editor1">
+      </div>
+    </div>
+    <div class="dataFieldContainer" [hidden]="dataFieldsVisible">
+      <div class="dataFieldHeaderContainer"></div>
+      <mat-expansion-panel>
+        <mat-expansion-panel-header>
+          <mat-panel-title>
+            Student
+          </mat-panel-title>
+        </mat-expansion-panel-header>
+        <ul id="dataFieldListStudent" class="dataFieldList">
+          <li *ngFor = "let dataField of filterDataFields('student')" [attr.data-dataFieldElement]="dataField.id" class="dataFieldElement" draggable="true">
```

Comment:
From a performance perspective, iterating over the result of a method call can have a negative impact because Angular will have to call that method continuously to check for updates. Could you pass a direct reference to that array instead? 
Perhaps create multiple sub-arrays from the start in Typescript (or a map) so that you don't need the filtering method anymore.

Reply:
Created an array for each type and removed filter method.